src/components/widgets/search-widget/search-widget.js

```diff
@@ -0,0 +1,62 @@
+// Docs for Search ui widget
+//https://developers.arcgis.com/javascript/latest/api-reference/esri-widgets-Search.html
+import { loadModules } from '@esri/react-arcgis';
+import React, { useEffect, useState } from 'react';
+import ReactDOM from 'react-dom';
+import SearchWidgetComponent from './search-widget-components';
+
+const SearchWidget = ({ view }) => {
+  const [searchWidget, setSearchWidget ] = useState();
+
+  const keyEscapeEventListener = (evt) => { 
+    evt = evt || window.event;
+    if (evt.keyCode == 27 && view && searchWidget) {
+      handleCloseSearch();
+    }
+  };
+
+  const handleOpenSearch = () => {
+    loadModules(["esri/widgets/Search"]).then(([Search]) => {
+      const sWidget = new Search({
+        view: view,
+        locationEnabled: false, // don't show the Use current location box when clicking in the input field
+        popupEnabled: false, // hide location popup
+        resultGraphicEnabled: false // hide location pin
+      });
+      setSearchWidget(sWidget);
+
+    }).catch((err) => console.error(err));
+  };
+
+  const handleCloseSearch = () => {
+    view.ui.remove(searchWidget);
+    document.removeEventListener('keydown', keyEscapeEventListener);
+    setSearchWidget(null);
+  }
+  
+  useEffect(() => {
+    if( searchWidget ) {
+      view.ui.add(searchWidget, "top-left");
+      document.addEventListener('keydown', keyEscapeEventListener);
+      searchWidget.viewModel.on("search-start", handleCloseSearch);
+      searchWidget.watch('activeSource', function(evt) {
+        evt.placeholder = "Search for a location";
+      });
+    }
+
+    const node = document.createElement("div");
+    view.ui.add(node, "top-left");
+    const component = <SearchWidgetComponent handleOpenSearch={handleOpenSearch} handleCloseSearch={handleCloseSearch} showCloseButton={!!searchWidget}/>;
+    ReactDOM.render(component, node);
```

Comment:
Maybe I'm missing something of the internals of `esri widgets` customization, but why are we creating a DOM node and adding the component to that node instead of returning it at the end of `SearchWidget` function?
And place it on the screen using `scss` instead of the Esri `ui` positions.
Not saying it has to be like this, just curious.

Reply:
:man_facepalming: :woman_facepalming:  I forgot how to React... fixed

Reply:
@kfilip94 I remember I asked you this before and you told me that it had the advantage of positioning it with esri's `top-left` that would always adjust to the panning and other components being rendered on this side of the screen. Not sure if it's still relevant.

Reply:
Yeah, an advantage of using `view.ui.add` and `ReactDOM.render` is that position of your component will be placed according to other esri widgets, so in this case, you don't have to specify CSS like  `top` or `left` properties. However, the `Find Places` button is always on the top-left corner of the viewport so setting a position with CSS is very easy. Also, `SearchWidgetComponent` is more complicated, the `Find Places` button depends on the `search widget` side effects and local state, so we have to also take care of removing `Find Places` button from the DOM.